### DISCUSSION TITLE
Using preregistered_runner_token_ssm_parameter_name causes registration failure on first deployment?

### DISCUSSION THREAD

**Discussion by akarl-bricks** [npalm/terraform-aws-gitlab-runner#1144] [Q&A]
If you use registration via the "preregistered_runner_token_ssm_parameter_name" parameter, the SSM parameter is created via Terraform and is hardcoded to "null". When the runner instance spins up, it fails to register (without error) because the parameter value is "null". You can obviously go set that parameter via console or other means but then you need to restart your runner instance.

After initial deployment, it seems fine, but this feels quite clunky for first time deployments. Am I missing something?

**Comment by kayman-mk** [comment]
Unfortunately you are not missing anything. You have to set up the SSM parameter, fill it with the correct token and then deploy this module here. Otherwise you will notice the failure described by you.

I do not see any chance to get it fixed as pre-register the Runner is always a manual task.
